SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestUnreliableRPCOverflow/SpatialTestUnreliableRPCOverflow.cpp

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestUnreliableRPCOverflow.h"
+#include "Kismet/GameplayStatics.h"
+#include "Net/UnrealNetwork.h"
+#include "SpatialFunctionalTestFlowController.h"
+#include "SpatialGDKSettings.h"
+#include "UnreliableRPCTestActor.h"
+
+/**
+ * This tests that when unreliable RPCs overflow, it will drop the older RPCs to sent the newer ones.
+ * Tests UNR-4360
+ * This test needs 1 Server and 1 Client workers.
+ *
+ * The flow is as follows:
+ * - Setup:
+ *  - Spawn an AUnreliableRPCTestActor in the server and give ownership to the Client
+ * - Test:
+ *  - Sent GetRPCRingBufferSize() + 2 unreliable RPCs from the server to the client.
+ *  - Check that the 2 oldest RPCs have been dropped, and the 2 most recent have been received.
+ */
+
+ASpatialTestUnreliableRPCOverflow::ASpatialTestUnreliableRPCOverflow()
+	: Super()
+	, RPCLimitCount(GetDefault<USpatialGDKSettings>()->GetRPCRingBufferSize(ERPCType::ClientUnreliable))
+{
+	Author = "Antoine Cordelle (antoine.cordelle@improbable.io)";
+	Description = TEXT("This tests that when unreliable RPCs overflow, it will drop the older RPCs to sent the newer ones.");
+}
+
+void ASpatialTestUnreliableRPCOverflow::PrepareTest()
+{
+	Super::PrepareTest();
+
+	AddStep(TEXT("The Server spawns one AUnreliableRPCTestActor and gives ownership to the Client"), FWorkerDefinition::Server(1), nullptr,
+			[this]() {
+				TestActor = GetWorld()->SpawnActor<AUnreliableRPCTestActor>(FVector(0.0f, 0.0f, 50.0f), FRotator::ZeroRotator,
+																			FActorSpawnParameters());
+				RegisterAutoDestroyActor(TestActor);
+
+				ASpatialFunctionalTestFlowController* Client1FlowController =
+					GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1);
+
+				AssertIsValid(TestActor, "Actor exists", this);
+				if (TestActor)
+				{
+					TestActor->SetReplicates(true);
+					TestActor->SetOwner(Client1FlowController->GetOwner());
+				}
+
+				FinishStep();
+			});
+
+	AddStep(TEXT("Make sure client has ownership of Actor"), FWorkerDefinition::Client(1), nullptr, nullptr, [this](float DeltaTime) {
+		if (IsValid(TestActor) && TestActor->GetOwner() == GetLocalFlowController()->GetOwner())
+		{
+			FinishStep();
+		}
+	});
```

Comment:
Not sure what happens by default if we don't set a time limit, but just in case, setting the initial ownership should probably succeed within 5 seconds.
```suggestion
	}, 5.0f);
```